### PR TITLE
feat: add bloom wallet

### PR DESCRIPTION
**Pull request by MarkNerdi**
### Description
This PR add Bloom wallet to the list of supported wallets

### Checklist
- [x] Increment the version field in `package.json` of the package you have made changes in following [semantic versioning](https://semver.org/) and using alpha release tagging
- [x] Check the box that allows repo maintainers to update this PR
- [x] Test locally to make sure this feature/fix works
- [x] Run `yarn check-all` to confirm there are not any associated errors
- [x] Confirm this PR passes Circle CI checks
- [x] Add or update relevant information in the documentation

### Docs Checklist
- [ ] Include a screenshot of any changes ([see docs README on running locally](https://github.com/blocknative/web3-onboard/blob/develop/docs/README.md))
- [x] Add/update the appropriate package README (if applicable)
- [x] Add/update the related module in the [docs demo](https://github.com/blocknative/web3-onboard/blob/develop/docs/src/lib/services/onboard.js) (if applicable)
- [x] Add/update the related package in the `docs/package.json` file (if applicable)

### If this PR includes changes to add an injected wallet or SDK wallet module: 
Please complete the following using the internal demo package.
To run this demo use the command `yarn && yarn dev` to get the project running at `http://localhost:8080/`

#### Tests with demo app (injected)
- [ ] send transaction
- [ ] switch chains
- [ ] sign message
- [ ] sign typed message
- [ ] disconnect

#### Tests with demo app (SDK)
- [x] send transaction
- [x] switch chains
- [x] sign message
- [x] sign typed message
- [x] disconnect

### CODE REVIEW

File: packages/bloom/package.json [hunk 1]
```diff
@@ -0,0 +1,77 @@
+{
+  "name": "@web3-onboard/bloom",
+  "version": "2.1.2",
```
Comment (Adamj1232) [inline packages/bloom/package.json:3]:
```suggestion
  "version": "2.0.0-alpha.1",
```

**Comment by Adamj1232** [comment]
Hi @MarkNerdi it looks like the build is failing within the PR status checks - can you have a look when you get a moment?

**Comment by MarkNerdi** [comment]
@Adamj1232 
Updated the code, can you run the workflow again pls? :pray:

**Comment by Adamj1232** [comment]
@MarkNerdi local testing and build went well. I formatted the files and added the bloom module to the internal demo - `packages/demo/package.json` where it wasnt added. Was bloom wallet functionality tested using the internal demo on your end?

**Comment by MarkNerdi** [comment]
@Adamj1232 
Yes i tested it locally, for that i had to remove `@web3onboad/bloom` from the package.json, as that isnt published yet

**Comment by Adamj1232** [comment]
@MarkNerdi code is looking good! When testing locally I am getting the following error:
Screenshot 2024-06-14 at 15 37 04

**Comment by MarkNerdi** [comment]
@Adamj1232 This error occurs when you dont have bloom installed on you machine. Do you have an example in your project where a wallet prompts the user to download it? i could add that

**Comment by Adamj1232** [comment]
> @Adamj1232 This error occurs when you dont have bloom installed on you machine. Do you have an example in your project where a wallet prompts the user to download it? i could add that

@MarkNerdi ah gotcha, yeah we would def want to display a download message so users can add Bloom if they want - here is an example to display that message formatted using a helper function - https://github.com/blocknative/web3-onboard/blob/eb205fb5356b6851afb764e42f06ac97b45052d2/packages/okx/src/index.ts#L35

**Comment by MarkNerdi** [comment]
@Adamj1232 
Sadly there is no consistent way to determine if an application is installed via the browser. The code in the newest PR checks if updating `location.href` to a specified protocl throws an error or not, which happens on Windows (and Mac) i think.

**Review by Adamj1232 — APPROVED** [review]
Looks good! Thanks for contributing to W3O!

**Pull request merged**